src/sentry/coreapi.py

```diff
@@ -691,6 +691,18 @@ def validate_data(self, project, data):
                 })
                 del data['release']
 
+        if data.get('dist'):
+            data['dist'] = six.text_type(data['dist'])
+            if not data.get('release'):
+                data['dist'] = None
+            elif len(data['dist']) > 64:
```

Comment:
Do we want to deal with special characters here now before we shoot ourselves in the foot again? Should we do other validation other than just simply 64 bytes? Can we restrict to like, `[a-z0-9-._]` or something. I'd rather not get us into the same problem trying to use these names as slugs in urls and having awful values.